tools/c7n_tencentcloud/c7n_tencentcloud/resources/ami.py

```diff
@@ -0,0 +1,83 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+import itertools
+
+from c7n.utils import type_schema
+from c7n.filters import Filter
+from c7n_tencentcloud.provider import resources
+from c7n_tencentcloud.query import ResourceTypeInfo, QueryResourceManager
+from c7n_tencentcloud.utils import PageMethod
+
+
+@resources.register("ami")
+class AMI(QueryResourceManager):
+    """ami"""
+
+    class resource_type(ResourceTypeInfo):
+        """resource_type"""
+        id = "ImageId"
+        endpoint = "cvm.tencentcloudapi.com"
+        service = "cvm"
+        version = "2017-03-12"
+        enum_spec = ("DescribeImages", "Response.ImageSet[]", {})
+        paging_def = {"method": PageMethod.Offset, "limit": {"key": "Limit", "value": 20}}
+        resource_prefix = "instance"
+        taggable = True
+
+    def get_resource_query_params(self):
+        """
+        https://cloud.tencent.com/document/api/213/15715
+        only query image-type = PRIVATE_IMAGE
+        """
+        config_query = self.data.get("query", [])
+        params = {
+            "Filters": [
+                {
+                    "Name": "image-type",
+                    "Values": ["PRIVATE_IMAGE"]
+                }
+            ]
+        }
+        for it in config_query:
+            params.update(it)
+
+        return params
+
+
+@AMI.filter_registry.register('unused')
+class ImageUnusedFilter(Filter):
+    """Filters images based on usage
+
+    true: image has no instances spawned from it
+    false: image has instances spawned from it
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ami-unused
+                resource: tencentcloud.ami
+                filters:
+                  - type: unused
+                    value: true
+    """
+    schema = type_schema('unused', value={'type': 'boolean'})
+
+    def get_permissions(self):
+        return list(itertools.chain(*[
+            self.manager.get_resource_manager(m).get_permissions()
+            for m in ('ami', 'cvm')]))
```

Comment:
nit: ami is redundant since the filter is always used by the ami resource manager